R/zchunk_L243.bio_trade_input.R

```diff
@@ -58,266 +60,404 @@ module_aglu_L243.bio_trade_input_DISABLED <- function(command, ...) {
     L120.LC_bm2_R_LT_Yh_GLU <- get_data(all_data, "L120.LC_bm2_R_LT_Yh_GLU")
     L102.pcgdp_thous90USD_Scen_R_Y <- get_data(all_data, "L102.pcgdp_thous90USD_Scen_R_Y")
 
-    # ===================================================
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses repeat_and_add_vector
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code converts gdp using a conv_xxxx_xxxx_USD constant
-    # Use the `gdp_deflator(year, base_year)` function instead
-    # ===================================================
+    # Note: I'm pulling out all of the hard-coded GCAM sector names used in this file to the top so it is clearer
+    OLD.REGIONAL.BIOMASS.NAME <- "regional biomass"
+    NEW.REGIONAL.BIOMASS.NAME <- "total biomass"
+    TRADED.BIOMASS.NAME <- "traded biomass"
+    BIOMASS.NAME <- "biomass"
+    BIOMASS.TRADE.REGION <- "USA"
+    DOMESTIC.BIOMASS.NAME <- "domestic biomass"
+    INTERNATIONAL.BIOMASS.NAME <- "imported biomass"
+
+    # First, create a table to delete existing regional biomass input (this needs to include region/sector/subsector/technology/input for all regions & years)
+    GCAM_region_names %>%
+      select(region) %>%
+      mutate(supplysector = OLD.REGIONAL.BIOMASS.NAME, subsector = OLD.REGIONAL.BIOMASS.NAME, technology = OLD.REGIONAL.BIOMASS.NAME) %>%
+      repeat_add_columns(tibble::tibble(year = MODEL_YEARS)) %>%
+      mutate(minicam.energy.input = BIOMASS.NAME) ->
+      L243.DeleteInput_RegBio
+
+    # Set up all of the new supply sectors.
+    # Copy sector information to each region.
+    # Note that "traded biomass" only goes in the BIOMASS.TRADE.REGION
+    A_bio_supplysector %>%
+      repeat_add_columns(tibble::tibble(region = GCAM_region_names$region)) %>%
+      mutate(region = if_else(traded == 1, BIOMASS.TRADE.REGION, region)) %>%
```

Comment:
how about `mutate(region = replace(region, traded == 1, BIOMASS.TRADE.REGION))`? 
If I remember correctly, `replace` is faster than `mutate + if_else`